### DISCUSSION TITLE
Github Actions Docker login

### DISCUSSION THREAD

**Discussion by sagikazarmark** [community/community#25520] [Actions]
<p>I’m trying to push to <a href="http://docker.pkg.github.com" rel="nofollow noopener">docker.pkg.github.com</a> from a GitHub action.</p>
<p>Here is the login step, which fails with 401 error:</p>
<pre><code>- name: Log in to registry
        uses: actions/docker/login@master
        env:
          DOCKER_REGISTRY_URL: docker.pkg.github.com
          DOCKER_USERNAME: ${{ github.actor }}
          DOCKER_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
        if: github.event_name == 'push'
</code></pre>
<p>Any ideas?</p>

**Comment by lee-dohm** [comment]
<p>I’m not sure that <code>github.actor</code> is the best choice for assigning the <code>DOCKER_USERNAME</code> value, unless that is always going to be someone with permissions to the package registry for that repository? What is <code>github.actor</code> set to when you get the 401 error?</p>

**Comment by bluk** [comment]
<p>I recently tried doing something similar.</p>
<p>It seems that the <code>GITHUB_TOKEN</code> does not have the correct access permissions (even though I believe it should based on <a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#token-permissions" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#token-permissions</a> ).</p>
<p>I generated a personal access token and added a repository encrypted secret to login. I’m also using the included Docker baked into the virtual image.</p>
<pre><code>- name: Login to GitHub Docker Registry
        run: docker login docker.pkg.github.com --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
        env:
          DOCKER_USERNAME: ${{ secrets.GITHUB_DOCKER_USERNAME }}
          DOCKER_PASSWORD: ${{ secrets.GITHUB_DOCKER_PASSWORD }}
</code></pre>

**Comment by sagikazarmark** [comment]
<p>I can’t really log <code>github.actor</code>, it’s filtered in the log.</p>
<p>What am I supposed to set the docker registry login when pushing to the GitHub registry?</p>
<p>If I have to manually create a personal access token (which I’m actually doing at the moment), that kinda defeats the whole purpose of using an integrated environment. Why is there a GITHUB_TOKEN secret then in the CI environment?</p>
<p>Also, there is a table in the documentation that says the GITHUB_TOKEN secret is a token with write permissions to packages:</p>
<p><a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#token-permissions" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#token-permissions</a></p>
<p>Is that not correct then?</p>

**Comment by sagikazarmark** [comment]
<p>I came to the same conclusion and did the same thing, but I think that’s quite a huge fail. If I want to use GitHub for CI and packages, I expect these services to be integrated. I kinda thought that was the value proposition in the first place: you can do everything in one place. Otherwise both GitHub Actions and GitHub Packages are “Yet Another Service”.</p>

**Comment by tomkins** [comment]
<p>Tried to do the same thing to get Actions creating Docker images locally - in the hope that having images stored in GitHub might be faster than having to fetch from Docker Hub.</p>
<p>However currently it feels like two different beta products aren’t quite ready for each other.</p>

**Comment by baldoalessandro** [comment]
<p>Yesterday I’ve tried to run <code>docker login ...</code> with a personal access token that’s working from my laptop and  I got this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a9c6b17a940b73d55c820887d5dddcd14c59c138.png" data-download-href="/uploads/short-url/odUoezvxO1iNPL8AACIjSwrbte8.png?dl=1" title="Screenshot from 2019-08-29 18-53-04.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/902985/181091702-74f68148-22b4-49d3-ad4d-29e3c17a9cf3.png" alt="Screenshot from 2019-08-29 18-53-04.png" data-base62-sha1="odUoezvxO1iNPL8AACIjSwrbte8" width="690" height="66" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/a/a9c6b17a940b73d55c820887d5dddcd14c59c138_2_690x66.png, https://aws1.discourse-cdn.com/github/optimized/2X/a/a9c6b17a940b73d55c820887d5dddcd14c59c138_2_1035x99.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/a/a9c6b17a940b73d55c820887d5dddcd14c59c138_2_1380x132.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a9c6b17a940b73d55c820887d5dddcd14c59c138_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot from 2019-08-29 18-53-04.png</span><span class="informations">1440×138 16.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>No idea why :slight_smile: I hope this will be fixed when “GitHub actions” leaves the beta stage</p>

**Comment by landsman** [comment]
<p>Same problem. This documentation about this is really weak :frowning:<br>
For me do not giving sense to create personal access here if there can be some global access based on access to repository.</p>

**Comment by kimtore** [comment]
<p>Struggling with the exact same thing now. One of the unique selling points for Github actions is seamless integration with other Github services - which is completely absent in this regard.</p>
<p>Please provide a no-hassle way to utilize package registry. All the configuration needed to do this already exists in the context (username and $GITHUB_TOKEN), so no need to boilerplate it in the YAML spec.</p>
<p>Maybe something in the lines of:</p>
<pre><code>- uses: actions/docker/login-package-registry@master
</code></pre>

**Comment by alfg** [comment]
<p>Same issue. Using GITHUB_TOKEN still results in a 401 when publishing using Github Actions.</p>

**Comment by ethomson** [comment, marked as answer]
<p>This is a known issue that we’re working to resolve.  GITHUB_TOKEN works for publishing most types of packages (eg, npm) but not yet for all.  This will work for all types of packages - including Docker - by the time GitHub Actions and GitHub Package Registry is out of beta.</p>

**Comment by riker09** [comment]
<p>I’ve had success with the following step:</p>
<pre><code>- name: Push
      run: |
        docker login docker.pkg.github.com --username RepoOwner --password "${GITHUB_PACKAGE_REGISTRY_TOKEN}"
        docker push docker.pkg.github.com/repoowner/repo_name/image_name:tag
      env:
        GITHUB_PACKAGE_REGISTRY_TOKEN: ${{ secrets.GITHUB_PACKAGE_REGISTRY_TOKEN }}
</code></pre>
<p>_ <strong>Note</strong> _: The Docker image name is <em>lowercase</em>. The username for docker login <em>may</em> contain uppercase letters. Make sure to convert them to lowercase when building the image tag string. “RepoOwner” -&gt; “repoowner”. Same goes for the repository name.</p>
<p>Here’s a full example: <a href="https://github.com/chrispat/actions-container-service/blob/master/.github/workflows/build-and-deploy.yml" rel="nofollow noopener">https://github.com/chrispat/actions-container-service/blob/master/.github/workflows/build-and-deploy.yml</a></p>

**Comment by rokcarl** [comment]
<p><a href="https://github.com/actions/docker" rel="nofollow noopener">https://github.com/actions/docker</a> repo is now a 404. Anyone know why?</p>

**Comment by nebularg** [comment]
<p>It has been deprecated for some time and they finally removed it.  Docker cli commands can be run directly from all the base images now so it wasn’t needed anymore.</p>

**Comment by fleskesvor** [comment]
<p>According to today’s announcement, GitHub package registry will leave beta on 13 November. Does that mean this issue has been (or is close to being) resolved?</p>

**Comment by fleskesvor** [comment]
<p>Can confirm that this works now.</p>

**Comment by codfish** [comment]
<p>When authenticating with the registry inside of GitHub Actions, do you need to provide a username along with the GITHUB_TOKEN? If so how can we do this programatically? Is there an “actions” user/actor and how can we access it? If <code>${{ github.actor }}</code> is the person who opened a PR or merged, etc. wouldn’t you expect to get auth errors when using a token that doesn’t belong to that actor?</p>
<p>In the same vein that we can just use the GITHUB_TOKEN, I dont want to have to use a personal account username here. Thoughts? Possible?</p>

**Comment by grepfruit19** [comment]
<p>Hello stranger,</p>
<p>It seems like you can use <code>${{ secrets.GITHUB_TOKEN }}</code> in conjunction with <code>${{ github.repository}}</code>, as in</p>
<p><code>docker login -u ${{ github.repostitory }} -p ${{ secrets.GITHUB_TOKEN }}</code></p>
<p>Edit: Never mind. While the login actually works, it doesn’t have the permissions needed to actually install the package…</p>
